src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -726,13 +726,32 @@ static int get_pool_offset(sd_dhcp_server *server, be32_t requested_ip) {
         return be32toh(requested_ip & ~server->netmask) - server->pool_offset;
 }
 
+static int prepare_new_lease(DHCPLease **lease, be32_t address, DHCPClientId *client_id, uint8_t *chaddr, be32_t gateway, usec_t expiration){
```

Comment:
we usually call return parameters ret_xyz, i.e. prefix them with `ret_` to indicate they are return parameters.